packages/react-router/__tests__/navigate-test.tsx

```diff
@@ -45,5 +45,152 @@ describe("<Navigate>", () => {
         </h1>
       `);
     });
+
+    it("handles upward navigation from an index routes", () => {
+      let renderer: TestRenderer.ReactTestRenderer;
+      TestRenderer.act(() => {
+        renderer = TestRenderer.create(
+          <MemoryRouter initialEntries={["/home"]}>
+            <Routes>
+              <Route path="home">
+                <Route index element={<Navigate to="../about" />} />
+              </Route>
+              <Route path="about" element={<h1>About</h1>} />
+            </Routes>
+          </MemoryRouter>
+        );
+      });
+
+      expect(renderer.toJSON()).toMatchInlineSnapshot(`
+        <h1>
+          About
+        </h1>
+      `);
+    });
+
+    it("handles upward navigation from inside a pathless layout route", () => {
+      let renderer: TestRenderer.ReactTestRenderer;
+      TestRenderer.act(() => {
+        renderer = TestRenderer.create(
+          <MemoryRouter initialEntries={["/home"]}>
+            <Routes>
+              <Route element={<Outlet />}>
+                <Route path="home" element={<Navigate to="../about" />} />
+              </Route>
+              <Route path="about" element={<h1>About</h1>} />
+            </Routes>
+          </MemoryRouter>
+        );
+      });
+
+      expect(renderer.toJSON()).toMatchInlineSnapshot(`
+        <h1>
+          About
+        </h1>
+      `);
+    });
+
+    it("handles upward navigation from inside multiple pathless layout routes + index route", () => {
+      let renderer: TestRenderer.ReactTestRenderer;
+      TestRenderer.act(() => {
+        renderer = TestRenderer.create(
+          <MemoryRouter initialEntries={["/home"]}>
+            <Routes>
+              <Route path="home">
+                <Route element={<Outlet />}>
+                  <Route element={<Outlet />}>
+                    <Route element={<Outlet />}>
+                      <Route index element={<Navigate to="../about" />} />
+                    </Route>
+                  </Route>
+                </Route>
+              </Route>
+              <Route path="about" element={<h1>About</h1>} />
+            </Routes>
+          </MemoryRouter>
+        );
+      });
+
+      expect(renderer.toJSON()).toMatchInlineSnapshot(`
+        <h1>
+          About
+        </h1>
+      `);
+    });
+
+    it("handles upward navigation from inside multiple pathless layout routes + path route", () => {
+      let renderer: TestRenderer.ReactTestRenderer;
+      TestRenderer.act(() => {
+        renderer = TestRenderer.create(
+          <MemoryRouter initialEntries={["/home/page"]}>
+            <Routes>
+              <Route path="home" element={<Outlet />}>
+                <Route element={<Outlet />}>
+                  <Route element={<Outlet />}>
+                    <Route element={<Outlet />}>
+                      <Route
+                        path="page"
+                        element={<Navigate to="../../about" />}
```

Comment:
@ryanflorence Quick double check here that this double `../..` would be the expected relative path to the `/about` route?